src/Mvc/test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/ValidationSummaryTagHelperTest.cs

```diff
@@ -587,7 +588,70 @@ public async Task ProcessAsync_GeneratesValidationSummaryWhenNotNone(ValidationS
                 expectedMessage);
         }
 
-        private static ViewContext CreateViewContext()
+        [Fact]
+        public async Task ProcessAsync_GeneratesExpectedOutput_WithModelErrorForIEnumerable()
```

Comment:
With only one error in `ModelState`, I doubt this test will fail without your fix. Please locally revert your changes to `ValidationHelpers` and test with
1. an _error_ for `FormMetadata.ID` (key is likely "ID" aka `nameof(FormMetadata.ID)`)
  - should be duplicated without your fix
2. that error and one for (say) `parameter`
  - should see just the error for `FormMetadata.ID` (twice) without your fix

Then, reapply your fix and confirm both tests display the expected errors without duplication.